platform/ios/src/MGLMapboxEvents.m

```diff
@@ -53,14 +51,29 @@ + (nullable instancetype)sharedInstance {
     return _sharedInstance;
 }
 
+// Readonly Reference to MMEEvents UserDefaults reference
++ (NSUserDefaults*)userDefaults {
+    return MMEEventsManager.sharedManager.configuration.userDefaults;
+}
+
+- (NSUserDefaults*)userDefaults {
+    return [self.class userDefaults];
+}
+
 - (instancetype)init {
     self = [super init];
     if (self) {
         _eventsManager = MMEEventsManager.sharedManager;
-        _eventsManager.debugLoggingEnabled = [[NSUserDefaults standardUserDefaults] boolForKey:MGLMapboxMetricsDebugLoggingEnabledKey];
+
+        if ([[NSUserDefaults standardUserDefaults] boolForKey :MGLMapboxMetricsDebugLoggingEnabledKey]) {
```

Comment:
Given the default is No in the defaults above, I wonder if this can migrate to be initialized and defaulted to None until a developer needs it?